File: Leetcode Challenge/February/Python/Compare Version Numbers.py
class Solution:
    def compareVersion(self, version1: str, version2: str) -> int:
        
        l1=list(version1.split("."))
        l2=list(version2.split("."))
      
        for i in range(max(len(l1),len(l2))):
            v1=0
            v2=0
            if i<len(l1):
                v1=int(l1[i])
            if i<len(l2):
                v2=int(l2[i])
                
            if v1<v2:
                return -1
            elif v2<v1:
                return 1
            
                
        return 0        
        

File: Leetcode Challenge/February/java/Compare Version Numbers.java
public class Solution {
public int compareVersion(String version1, String version2) {
    int temp1 = 0,temp2 = 0;
    int len1 = version1.length(),len2 = version2.length();
    int i = 0,j = 0;
    while(i<len1 || j<len2) {
        temp1 = 0;
        temp2 = 0;
        while(i<len1 && version1.charAt(i) != '.') {
            temp1 = temp1*10 + version1.charAt(i++)-'0';
            
        }
        while(j<len2 && version2.charAt(j) != '.') {
            temp2 = temp2*10 + version2.charAt(j++)-'0';
            
        }
        if(temp1>temp2) return 1;
        else if(temp1<temp2) return -1;
        else {
            i++;
            j++;
            
        }
        
    }
    return 0;
    
}
}
